src/pip/_internal/index/package_finder.py

```diff
@@ -552,9 +564,27 @@ def compute_best_candidate(
         """
         Compute and return a `BestCandidateResult` instance.
         """
-        applicable_candidates = self.get_applicable_candidates(candidates)
 
-        best_candidate = self.sort_best_candidate(applicable_candidates)
+        # Apply priority filtering to the list of candidates
+        found_candidate = False
+        if(self._index_priority and self._index_urls is not None):
```

Comment:
This is a good example of the opt-in index priority described by PEP 766. I think this is ultimately not a good approach, because it forces a behavior change away from Pip. I think that is going to be disruptive enough that it will be very hard to get into Pip. The behavior you have here should be similar to uv's behavior.